Deploy and Manage: ECK (#380)
elastic/docs-content: deploy-manage/deploy/cloud-on-k8s/install-using-yaml-manifest-quickstart.md

% Use migrated content from existing pages that map to this page:

% removed both
% - [ ] ./raw-migrated-files/cloud-on-k8s/cloud-on-k8s/k8s-install-yaml-manifests.md
% - [ ] ./raw-migrated-files/cloud-on-k8s/cloud-on-k8s/k8s-deploy-eck.md

> This method is the quickest way to get started with ECK if you have full administrative access to the Kubernetes cluster. 

Reviewer: ```suggestion
In this guide, you'll learn how to deploy ECK using Elastic-provided YAML manifests. This method is the quickest way to get started with ECK if you have full administrative access to the Kubernetes cluster. 

To learn about other installation methods, refer to [](/deploy-manage/deploy/cloud-on-k8s/install.md).
```

During the installation, the following components are installed or updated:

* `CustomResourceDefinition` objects for all supported resource types (Elasticsearch, Kibana, APM Server, Enterprise Search, Beats, Elastic Agent, Elastic Maps Server, and Logstash).
* `Namespace` named `elastic-system` to hold all operator resources.
* `ServiceAccount`, `ClusterRole` and `ClusterRoleBinding` to allow the operator to manage resources throughout the cluster.
* `ValidatingWebhookConfiguration` to validate Elastic custom resources on admission.
* `StatefulSet`, `ConfigMap`, `Secret` and `Service` in `elastic-system` namespace to run the operator application.

## Prerequisites and considerations

Before you begin, review the following prerequisites and recommendations:

> * For this quickstart guide, your Kubernetes cluster is assumed to be already up and running. Before you proceed with the ECK installation, make sure you check the [supported versions](/deploy-manage/deploy/cloud-on-k8s.md#k8s-supported).

Reviewer: could simplify this. is there a word that applies to this group of items? "operator"? "orchestrator"? "platform"?

```suggestion
* You're running a Kubernetes cluster using a [supported platform](/deploy-manage/deploy/cloud-on-k8s.md#k8s-supported).
```

* If you are using GKE, make sure your user has `cluster-admin` permissions. For more information, check [Prerequisites for using Kubernetes RBAC on GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control#iam-rolebinding-bootstrap).

> * If you are using Amazon EKS, make sure the Kubernetes control plane is allowed to communicate with the Kubernetes nodes on port 443. This is required for communication with the Validating Webhook. For more information, check [Recommended inbound traffic](https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.md).

Reviewer: ```suggestion
* If you are using Amazon EKS, make sure the Kubernetes control plane is allowed to communicate with the Kubernetes nodes on port 443. This is required for communication with the validating webhook. For more information, check [Recommended inbound traffic](https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html).
```

> * Refer to [*Install ECK*](../../../deploy-manage/deploy/cloud-on-k8s/install.md) for more information on installation options.
> 
> * Check the [upgrade notes](../../../deploy-manage/upgrade/orchestrator/upgrade-cloud-on-k8s.md) if you are attempting to upgrade an existing ECK deployment.

Reviewer: ```suggestion
```

##  Installation procedure

To deploy the ECK operator:

> 1. Install [custom resource definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) with [`create`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_create/):

Reviewer: ```suggestion
1. Install Elastic's [custom resource definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) with [`create`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_create/):
```

    ```sh
    kubectl create -f https://download.elastic.co/downloads/eck/{{eck_version}}/crds.yaml

>     ```

Reviewer: this variable `{{eck_version}}` does not exist in `docset.yml` and needs to be added.

not sure if it will display after it's added - if it doesn't, we'll need to log it as a bug. according to martijn [it should work fine](https://elastic.slack.com/archives/C05PP2LEC1X/p1739387960361149)

Author: making note of it in the parent issue. left for the next PR.

>     This will output similar to the following upon Elastic resources' creation:

Reviewer: ```suggestion
    You'll see output similar to the following as resources are created:
```

    ```sh
    customresourcedefinition.apiextensions.k8s.io/agents.agent.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/apmservers.apm.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/beats.beat.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/elasticmapsservers.maps.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/elasticsearches.elasticsearch.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/enterprisesearches.enterprisesearch.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/kibanas.kibana.k8s.elastic.co created
    customresourcedefinition.apiextensions.k8s.io/logstashes.logstash.k8s.elastic.co created
    ```

> 2. Install the operator with its RBAC rules with [`apply`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_apply/):

Reviewer: ```suggestion
2. Using [`kubectl apply`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_apply/), install the operator with its RBAC rules:
```

    ```sh
    kubectl apply -f https://download.elastic.co/downloads/eck/{{eck_version}}/operator.yaml
    ```

    ::::{note}

>     The ECK operator runs by default in the `elastic-system` namespace. It is recommended that you choose a dedicated namespace for your workloads (such as Elasticsearch and Kibana), rather than using the `elastic-system` or the `default` namespace.

Reviewer: the brackets don't make sense to me. do you mean the person should create an `elasticsearch` and `kibana` namespace? a dedicated namespace for elasticsearch and kibana? how is "workloads" defined?

Author: workloads = applications (in this context orchestrated by the operator) = Elastic Stack apps = Elasticsearch and Kibana (among others of course).

I added the brackets for people that might not be familiar with the kubernetes term of `workloads`. But I'll remove it and leave the original text. The entire note doesn't add much value actually.

Maybe we should just mention to not orchestrate applications in `elastic-system` namespace instead.

Author: I've updated it to:

```
    ::::{note}
    The ECK operator runs by default in the `elastic-system` namespace. While this namespace is used for the operator itself, it is recommended that you deploy your application workloads in a separate, dedicated namespace instead of `elastic-system` or `default`. You will need to consider this when setting up your applications.
    ::::
```

I think it's much clearer now :)

    ::::

> 3. Monitor the operator’s setup from its logs through [`logs`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_logs/):

Reviewer: prefer prefacing the "tool" to use and then state the action.

this feels optional. consider adding this as a paragraph to the previous step. if that's the case, "to monitor the installation process, use [command] ..."

```suggestion
3. Using [`kubectl logs`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_logs), monitor the operator’s setup by watching the logs:
```

    ```sh
    kubectl -n elastic-system logs -f statefulset.apps/elastic-operator
    ```

> 4. Once ready, the operator will report as `Running` as shown with [`get`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_get/), replacing default `elastic-system` with applicable installation namespace as needed: *

Reviewer: ```suggestion
4. Use [`kubectl get`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_get/) to check the operator status, passing in the namespace using the `-n` flag. When the operator is ready to use, it will report as `Running`. 
```

```
$ kubectl get -n elastic-system pods
NAME                 READY   STATUS    RESTARTS   AGE
elastic-operator-0   1/1     Running   0          1m

> ```

Reviewer: this should be indented, and split into a copyable command and response (two separate codeblocks) - could separate by the "When the operator is ready to use, it will report as `Running`. " sentence

> This completes the quickstart of the ECK operator. We recommend continuing to [Deploying an {{es}} cluster](../../../deploy-manage/deploy/cloud-on-k8s/elasticsearch-deployment-quickstart.md); but for more configuration options as needed, navigate to [Operating ECK](../../../deploy-manage/deploy/cloud-on-k8s/configure.md).

Reviewer: especially in the quickstart area this next steps heading can keep people moving forward

```suggestion
## Next steps

* To continue the quickstart, go to [](/deploy-manage/deploy/cloud-on-k8s/elasticsearch-deployment-quickstart.md)
* For more configuration options, refer to [](/deploy-manage/deploy/cloud-on-k8s/configure.md).
```
